posthog/hogql_queries/hogql_query_runner.py

```diff
@@ -66,3 +66,16 @@ def _is_stale(self, cached_result_package):
 
     def _refresh_frequency(self):
         return timedelta(minutes=1)
+
+    def apply_dashboard_filters(self, dashboard_filter: DashboardFilter) -> HogQLQuery:
+        self.query.filters = self.query.filters or HogQLFilters()
+        self.query.filters.dateRange = self.query.filters.dateRange or DateRange()
+
+        if dashboard_filter.date_to or dashboard_filter.date_from:
+            self.query.filters.dateRange.date_to = dashboard_filter.date_to
+            self.query.filters.dateRange.date_from = dashboard_filter.date_from
+
+        if dashboard_filter.properties:
+            self.query.filters.properties = (self.query.filters.properties or []) + dashboard_filter.properties
+
+        return self.query
```

Comment:
I wonder if we actually should mutate `self.query` here. It could be an object that's passed to the runner without any expectation that it's going to change. 

Perhaps it's thus "cleaner" to save the filters to `self._dashboard_filter = dashboard_filter`, and merge it on the fly in `to_query`?

Alternatively, we could just store this instance variable in a generic function `query_runner.py`, and expose something like `get_dashboard_filters`, which each query runner calls at the right time to fetch them down, if any... 🤔